blocks/api/factory.js

```diff
@@ -7,26 +7,30 @@ import { get, castArray, findIndex, isObjectLike, find } from 'lodash';
 /**
  * Internal dependencies
  */
-import { getBlockSettings } from './registration';
+import { getBlockType } from './registration';
 
 /**
- * Returns a block object given its type and attributes
+ * Returns a block object given its type and attributes.
  *
- * @param  {String} blockType   BlockType
+ * @param  {String} blockName   Block name
  * @param  {Object} attributes  Block attributes
  * @return {Object}             Block object
  */
-export function createBlock( blockType, attributes = {} ) {
-	const blockSettings = getBlockSettings( blockType );
+export function createBlock( blockName, attributes = {} ) {
+	// Get the type definition associated with a registered block.
+	const blockType = getBlockType( blockName );
```

Comment:
The purpose of the PR is precisely to use `blockType` as the bespoke "prototype" or type description of the block, not the name, not the block instance.